doc/sphinx/compiling.rst

```diff
@@ -32,69 +33,267 @@ Installation Prerequisites
 Linux
 -----
 
-* For Ubuntu or Debian users, the following packages should be installed using
+General Notes
+^^^^^^^^^^^^^
+
+* To download the source code, installing ``git`` is highly recommended.
+
+* SCons is only available for Python 2, so building the Python 3 module requires
+  two installations of Python (one of Python 2 and one of Python 3), even if you
+  do not intend to build the Python 2 module.
+
+* The following instructions use the system-installed versions of Python, but
+  alternate installations such as the Anaconda distribution of Python can be
+  used as well.
+
+* Cython is only required to be installed for the version of Python that also
+  has SCons installed; following the instructions below will install Cython for
+  the version of Python 2 installed in the system directories.
+
+* The SUNDIALS suite of solvers used in Cantera is available from several sources.
+  The sources in most distribution repositories may be out of date, or not
+  optimized for your hardware. The source code for SUNDIALS will be downloaded
+  and compiled with the rest of Cantera, so installing from your distribution's
+  repositories is optional.
```

Comment:
I guess I mean minimal for each interface. For instance, my recommended installs for the Python interface would also include IPython and Matplotlib, but I don't think that those belong in this section, because they're not necessary to compile Cantera, just as a system install of SUNDIALS is not necessary to compile Cantera. Nonetheless, I take your point about the inherent discrepancy in the build process as its currently constructed, and I agree that it shouldn't be changed without more discussion (i.e., after 2.3.0). (FWIW, I think I prefer the current setup where the builder searches for system installs by default, since there is an option to disable that search).

What I'm trying to document here is what you absolutely need to install to build Cantera from a totally fresh OS install. If you have other packages already installed, so be it, the builder will try to find those, but they're not strictly required. Does it make sense to do that?